src/core/os.c

```diff
@@ -497,6 +498,45 @@ static Janet os_cwd(int32_t argc, Janet *argv) {
     return janet_cstringv(ptr);
 }
 
+static Janet os_cryptorand(int32_t argc, Janet *argv) {
+    janet_fixarity(argc, 1);
+    const char *errmsg = "unable to get sufficient random data";
+    JanetBuffer *buffer = janet_getbuffer(argv, 0);
+#ifdef JANET_WINDOWS
+    HCRYPTPROV ctx;
+    if (!CryptAcquireContext(&ctx, NULL, NULL, PROV_RSA_FULL, CRYPT_VERIFYCONTEXT))
+        janet_panic(errmsg);
+    if (!CryptGenRandom(ctx, buffer->count, buffer->data)) {
+        CryptReleaseContext(ctx, 0);
+        janet_panic(errmsg);
+    }
+    if (!CryptReleaseContext(ctx, 0))
+        janet_panic(errmsg);
+#elif defined(__linux__) || defined(__APPLE__)
+    /* We should be able to call getrandom, but it doesn't seem  to be uniformly supported
+       on linux distros... */
+    int randfd = open("/dev/urandom", O_RDONLY);
+    if (randfd < 0)
+        janet_panic(errmsg);
+    size_t remaining = buffer->count;
+    while (remaining) {
+        ssize_t n = read(randfd, buffer->data + (remaining - buffer->count), remaining);
+        if (n <= 0) {
+            close(randfd);
```

Comment:
@bakpakin To be honest, I'm not sure what to do in the case of EINTR on these calls, especially close, if its ok to just ignore them.